app/src/main/java/org/thoughtcrime/securesms/ui/components/QrImage.kt

```diff
@@ -0,0 +1,104 @@
+package org.thoughtcrime.securesms.ui.components
+
+import android.graphics.Bitmap
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.animation.fadeIn
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.aspectRatio
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.Card
+import androidx.compose.material.Icon
+import androidx.compose.material.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.graphics.asImageBitmap
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import network.loki.messenger.R
+import org.thoughtcrime.securesms.ui.LocalLightCell
+import org.thoughtcrime.securesms.ui.LocalOnLightCell
+import org.thoughtcrime.securesms.util.QRCodeUtilities
+
+@Composable
+fun QrImage(
+    string: String,
+    contentDescription: String,
+    modifier: Modifier = Modifier,
+    icon: Int = R.drawable.session_shield
+) {
+    var bitmap: Bitmap? by remember {
+        mutableStateOf(null)
+    }
+
+    val scope = rememberCoroutineScope()
+    LaunchedEffect(string) {
+        scope.launch(Dispatchers.IO) {
+            val c = 100
```

Comment:
What's all this?

Reply:
It was a bad way of cutting out a square in the middle of the QR, I'll do something better.

Reply:
done